plugins/ProfileExtender/class.profileextender.plugin.php

```diff
@@ -318,7 +318,7 @@ public function settingsController_profileFieldAddEdit_create($Sender, $Args) {
                 // Make unique name from label for new fields
                 $Name = $TestSlug = substr(preg_replace('`[^0-9a-zA-Z]`', '', val('Label', $FormPostValues)), 0, 50);
                 $i = 1;
-                while (array_key_exists($Name, $Fields) || in_array($Name, $this->ReservedNames)) {
+                while (empty($Name) || array_key_exists($Name, $Fields) || in_array($Name, $this->ReservedNames)) {
```

Comment:
If you have an empty `$Name` that means `$TestSlug` is also empty, which means on the next line you are allowing it to just be a numeric `$Name`, no? That seems like a problem.